### DISCUSSION TITLE
[ Section 3 - 13 Interpolação, locals e outputs]

### DISCUSSION THREAD

**Discussion by fabioffalcao** [chgasparoto/curso-aws-com-terraform#66] [Q&A]
Quando devo usar variáveis (arquivos variables.tf e tfvars) e quando devo usar o locals?
Existe algum padrão que devemos seguir ou tanto faz?

**Comment by chgasparoto** [comment]
@fabioffalcao o arquivo de **_variables.tf_** é para você definir as suas variáveis, como o tipo, descrição e valor default. O arquivo **_tfvars_** é para você apenas setar os valores das variáveis que você definiu no outro arquivo.

O **_locals_** é bem útil para quando você tem que fazer algum cálculo (interpolação, função, etc) dentro dos arquivos do Terraform e usar em mais de um lugar/arquivo.
